src/chainstate/stacks/db/accounts.rs

```diff
@@ -0,0 +1,829 @@
+                    &true as &dyn ToSql, &0i64 as &dyn ToSql])
+            .map_err(|e| Error::DBError(db_error::SqliteError(e)))?;
+
+        for user_support in user_burns.iter() {
+            tx.execute("INSERT INTO payments (address,block_hash,burn_header_hash,parent_block_hash,parent_burn_header_hash,coinbase,tx_fees_anchored,tx_fees_streamed,stx_burns,burnchain_commit_burn,burnchain_sortition_burn,fill,stacks_block_height,miner,vtxindex) \
+                        VALUES (?1,?2,?3,?4,?5,?6,?7,?8,?9,?10,?11,?12,?13,?14,?15)",
+                        &[&user_support.address.to_string(), &block_reward.block_hash.to_hex(), &block_reward.burn_header_hash.to_hex(), &block_reward.parent_block_hash.to_hex(), &block_reward.parent_burn_header_hash.to_hex(),
+                        &format!("{}", block_reward.coinbase), &"0".to_string(), &"0".to_string(), &"0".to_string(),
+                        &(user_support.burn_amount as i64) as &dyn ToSql, &(block_reward.burnchain_sortition_burn as i64) as &dyn ToSql, &format!("{}", block_reward.fill), &(block_reward.stacks_block_height as i64) as &dyn ToSql,
+                        &false as &dyn ToSql, &user_support.vtxindex as &dyn ToSql])
+                .map_err(|e| Error::DBError(db_error::SqliteError(e)))?;
+        }
+
+        Ok(())
+    }
+ 
+    /// Get the scheduled miner rewards in a particular Stacks fork at a particular height
+    fn get_scheduled_block_rewards_in_fork<'a>(tx: &mut StacksDBTx<'a>, tip: &StacksHeaderInfo, block_height: u64) -> Result<Vec<MinerPaymentSchedule>, Error> {
+        let ancestor_info = match StacksChainState::get_tip_ancestor(tx, tip, block_height)? {
+            Some(info) => info,
+            None => {
+                return Ok(vec![]);
+            }
+        };
+        
+        let row_order = MinerPaymentSchedule::row_order().join(",");
+        let qry = format!("SELECT {} FROM payments WHERE block_hash = ?1 AND burn_header_hash = ?2 ORDER BY vtxindex ASC", row_order);
+        let args = [&ancestor_info.anchored_header.block_hash().to_hex(), &ancestor_info.burn_header_hash.to_hex()];
+        let rows = query_rows::<MinerPaymentSchedule, _>(tx, &qry, &args).map_err(Error::DBError)?;
+        Ok(rows)
+    }
+
+    /// Calculate the total reward for a miner (or user burn support), given a sample of scheduled miner payments.
+    /// The scheduled miner payments must be in order by block height (sample[0] is the oldest).
+    /// The first tuple item is the miner's reward; the second tuple item is the list of
+    /// user-support burns that helped the miner win.
+    ///
+    /// There must be MINER_REWARD_WINDOW items in the sample.
+    ///
+    /// TODO: .burns?
+    fn calculate_miner_reward(miner: &MinerPaymentSchedule, sample: &Vec<(MinerPaymentSchedule, Vec<MinerPaymentSchedule>)>) -> MinerReward {
```

Comment:
Don't bother reading this method for now -- it is broken to the point where it will need to be rewritten.  Sorry for not catching that sooner!